runtime-sdk/src/modules/core/mod.rs

```diff
@@ -11,38 +11,31 @@ pub mod types;
 /// Unique module name.
 pub const MODULE_NAME: &str = "core";
 
-// TODO: Add a custom derive macro for easier error derivation (module/error codes).
 /// Errors emitted by the core module.
-#[derive(Error, Debug)]
+#[derive(Error, Debug, oasis_runtime_sdk_macros::Error)]
+#[runtime_error(module_name_path = "MODULE_NAME")]
```

Comment:
Could it be `sdk::error` or something? Also this seems slightly inconsistent with how it's done for events? Could we use the same syntax for both.

Reply:
Unfortunately it needs to be an `Ident`. `runtime_event` sounds sad, though. Let me mess with it a bid.